pandas/tests/test_index.py

```diff
@@ -938,6 +938,10 @@ def test_difference(self):
         self.assertEqual(len(result), 0)
         self.assertEqual(result.name, first.name)
 
+        # empty difference for subtypes
+        result = self.periodIndex.difference(self.periodIndex)
+        self.assertIsInstance(result, pd.PeriodIndex)
+
     def test_symmetric_diff(self):
```

Comment:
Actually move this test [here](https://github.com/pydata/pandas/blob/master/pandas/tests/test_index.py#L338)

this will tests ALL index types. So you may see other things fall out (as in broken).